lib/ansible/modules/clustering/kubevirt/kubevirt_cdi_upload.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2018, KubeVirt Team <@kubevirt>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+
+module: kubevirt_cdi_upload
```

Comment:
I think we should consider adding just single module for all CDI functionality.

Reply:
I thought so initially as well, but later figured it'd make `kubevirt_pvc` more complicated that way and a bit harder to use, as it'd need to internally do many different things at once. Splitting the functionality, I get a clean idempotent `kubevirt_pvc` and a small dedicated upload module, which works fine without `kubevirt_pvc`, simply because you can create a CDI–ready PVC with just `k8s` module, but you can't actually upload data without dedicated code.

Reply:
Ok, when `kubevirt_pvc` is done, we can re-thing the integration.